engine/runtime/src/main/java/org/enso/compiler/TreeToIr.java

```diff
@@ -947,6 +958,30 @@ yield switch (ast) {
     */
   }
 
+  private IR$Application$Prefix translateAnnotation(IR$Name$Annotation ir, Tree expr, List<IR.CallArgument> callArgs) {
+    boolean insideTypeSignature = false;
+    return switch (expr) {
+      case Tree.App fn -> {
+        var fnAsArg = translateCallArgument(fn.getArg(), insideTypeSignature);
+        yield translateAnnotation(ir, fn.getFunc(), cons(fnAsArg, callArgs));
+      }
+      case Tree.ArgumentBlockApplication fn -> {
+        var fnAsArg = translateCallArgument(fn.getLhs(), insideTypeSignature);
+        var arg = translateCallArgument(expr, insideTypeSignature);
+        callArgs = cons(fnAsArg, cons(arg, callArgs));
+        yield translateAnnotation(ir, null, callArgs);
+      }
+      case null -> {
```

Comment:
That is a bit surprising case

Reply:
Line 972 and line 980 reach it via `yield translateAnnotation(ir, null, callArgs);`